src/main/java/com/vaadin/guice/server/GuiceViewProvider.java

```diff
@@ -50,19 +52,20 @@
 @Singleton
 class GuiceViewProvider implements ViewProvider, SessionDestroyListener, SessionInitListener {
 
+    private static final long serialVersionUID = 6113953554214462809L;
+
     private final Map<String, Class<? extends View>> viewNamesToViewClassesMap;
     private final Map<VaadinSession, Map<String, View>> viewsBySessionMap;
-    private final Set<String> viewNames;
+    private final NavigableSet<String> viewNames;
 
     public GuiceViewProvider(Set<Class<? extends View>> viewClasses) {
 
         viewNamesToViewClassesMap = scanForViews(viewClasses);
-        viewNames = viewNamesToViewClassesMap.keySet();
+        viewNames = ImmutableSortedMap.copyOf(viewNamesToViewClassesMap).keySet();
```

Comment:
Added some comments in the code. The purpose of the line is merely to get a sorted/navigable set of view names.